core/templates/dev/head/domain/topic/TopicUpdateService.js

```diff
@@ -205,15 +214,20 @@ oppia.factory('TopicUpdateService', [
       },
 
       /**
-       * Deletes a subtopic from the topic and records the change in
-       * the undo/redo service.The argument isNewlyCreated should be passed from
-       * the calling function, after checking the changelist to see if this
-       * subtopic was created in the same changelist.
+       * @param {Topic} topic - The topic object to be edited.
+       * @param {number} subtopicId - The id of the subtopic to delete.
+       * @param {boolean} isNewlyCreated - Whether the subtopic to delete was
+       *    created in the current draft of the topic. (i.e, the subtopic to
+       *    delete hasn't been saved in the datastore yet.)
        */
       deleteSubtopic: function(topic, subtopicId, isNewlyCreated) {
         var subtopic = topic.getSubtopicById(subtopicId);
+        if (!subtopic) {
+          return false;
```

Comment:
For all the various "return false" stuff, would it be better to `throw Error(...)` instead, if these are not expected to arise in practice?

Reply:
Done.